### ISSUE TITLE
Generic Git repo: calls to webhook require headers and payload

### ISSUE THREAD

**Issue by gdl-reif** [Dokploy/dokploy#633]
### To Reproduce

1. Deploy an app from a Git repository source (I used a repo on Azure DevOps and a matching SSH key to deploy manually, which worked fine)
2. Copy webhook url
3. Call webhook url from a browser (GET) or using Postman (POST)

### Current vs. Expected behavior

**Current:** "branch not match"

**Expected:** deploy app with it's settings from Dokploy (i.e. configured branch and last commit) as generic Git repository providers don't send any payload, or not a payload that Dokploy can interpret

### Provide environment information

```bash
Operating System: Ubuntu 24.04.1 LTS
Dokploy: 0.10.4
VPS Provider: Azure
App: Dotnet backend+Svelte frontend using a custom Dockerfile which is in the repo root
```

### Which area(s) are affected? (Select all that apply)

Application

### Additional context

If I pass a fake "X-Event-Key" header and fake JSON body along with the request, it works: 

```
X-Event-Key: repo:push

{ 
    "push" : {
        "changes" : [
            { 
                "new" : { "name" : "master", "target" : { "message" : "Auto deploy", "hash" : "NEW_COMMIT" } }
            }
        ]
    }
}
```

**Comment by Siumauricio** [comment]
This is the expected behavior, we support these Git providers for auto deploy https://docs.dokploy.com/en/docs/core/application/auto-deploy#webhook-url

We do not have a generic webhook because the body varies depending on each provider, What we could do is add this new provider to the list of supported webhooks

**Comment by gdl-reif** [comment]
@Siumauricio Thanks for the clarification! IMO matching the webhook to the git provider is perfect when I select a source git repo from one of those specific providers. But if I select a generic Git repo (see below), my expectation was to also get a generic webhook (i.e. one without special requirements).

![image](https://github.com/user-attachments/assets/5b98d010-f9b9-40b0-ba34-3d8a1a9ed7a7)

Of course, the better solution would be to integrate Azure DevOps into Dokploy, if that's possible. The payload the POST webhook sends should be pretty straightforward to parse. I'd be glad to supply information and help out where I can.

Sample Payload (Azure will not send headers, as far as I could tell):
[payload.json.txt](https://github.com/user-attachments/files/17584709/payload.json.txt)

**Comment by gdl-reif** [comment]
PS: I've looked into the API way to trigger the process, and will probably integrate that first. Thanks for pointing out that alternative, I think it's pretty universal. As we're aiming to auto-create apps on the fly for every PR and auto-deploy them, I think we'll dig into the API anyways.

**Comment by Siumauricio** [comment]
Yeah I know the ideal should be to autodeploy for any repo however we need to validate the branches in order to match the branch you specified in dokploy, you can also take a look this the usage of API to trigger deployments probably this can solve your issue for now, let me know what you think!

https://github.com/Dokploy/dokploy/issues/164#issuecomment-2190980695

**Comment by gdl-reif** [comment]
Thanks for the link! I'll try to work out a custom build step to make the API call. If there's ever any intent to tightly integrate Azure DevOps, I'd love to help out (or maybe submit a PR for that "some time in the future").

**Comment by Kansuler** [comment]
An addition to this issue is in case you call the webhook from a release/tag, then ref becomes `"ref": "refs/tags/v1.0.11"` which is changed between webhook calls. I'm currently considering how to work around this issue.

**Comment by shahabhm** [comment]
Hi @Siumauricio 

Is there a plan to add github to the supported git providers for autodeploy?

I have a use case for it explained here: [link](https://github.com/Dokploy/dokploy/issues/2149#issuecomment-3263805988) [link](https://github.com/Dokploy/dokploy/issues/2149#issuecomment-3263810858)

**Comment by dev7ch** [comment]
... and webhook deploy trigger requires autodeploy to be enabled, i still get not behind why this is useful and in which case this is needed.

Does anyone know why?

I d expect if autodeploy disabled webhook should still work
